javacs-lab12/src/com/flatironschool/javacs/WikiSearch.java

```diff
@@ -159,13 +176,83 @@ public static WikiSearch search(String term, JedisIndex index) {
 		return new WikiSearch(map);
 	}
 
+	public static WikiSearch searchTFIDF(String term, JedisIndex index) {
+		if(index.getCountsTFIDF(term) == null) { return null; }
+		Map<String, Integer> map = index.getCountsTFIDF(term);
+		return new WikiSearch(map);
+	}
+
+
+	/* Index all of the pages locally stored in the resources folder
+	    	- only need to index once */
+	public static void indexResourcesFolder(JedisIndex index) throws IOException {
+		/*index.deleteURLSets();
+		index.deleteAllKeys();
+		index.deleteTermCounters();*/
+
+		WikiFetcher wf = new WikiFetcher();
+		for(Entry<String, Elements> entry : wf.readAllWikipedia().entrySet()) {
+			index.indexPage(entry.getKey(), entry.getValue());
+		}
+	}
+
+
+
 	public static void main(String[] args) throws IOException {
 
 		// make a JedisIndex
 		Jedis jedis = JedisMaker.make();
 		JedisIndex index = new JedisIndex(jedis);
 
-		// search for the first term
+
+
+		/* Index all of the wiki pages in the resource folder */
+		//indexResourcesFolder(index);
+
+
+
+		/* TF - IDF */
+		List<String> queryList = new ArrayList<String>();
+		queryList.addAll(Arrays.asList("java", "and", "the", "philosophy", "currently"));
+
+			for(String query : queryList) {
+				/*If the query has a document with a negative TF-IDF score (searchTFIDF() returns null),
+					cease searching on that query - that word is a stop word */
+				if(searchTFIDF(query, index) == null) {
+					System.out.println("Ignored Query: " + query);
+					continue;
+				} else {
+					System.out.println("Query: " + query);
+					WikiSearch search = searchTFIDF(query, index);
```

Comment:
now calling searchTFIDF twice leads to calling getCountsTFIDF 4 times which calls getTFIDF 8 times and so on